zerver/tests/test_messages.py

```diff
@@ -2749,11 +2755,22 @@ def test_get_first_visible_message_id_with_limit(self) -> None:
         realm.save()
 
         expected_message_id = message_ids[5]
+        with mock.patch('zerver.lib.message.estimate_recent_messages', return_value=15):
+            maybe_update_first_visible_message_id(realm, 30)
         self.assertEqual(get_first_visible_message_id(realm), expected_message_id)
+
+        # If the number of messages sent in lookback_period is 0
+        # then the cache value should not be updated
```

Comment:
And this part should be pulled out into a unit test specifically for the "maybe" logic. That test can mock out the update function, so it's just checking directly whether that function gets called or not.